[13.0][IMP] stock_picking_group_by_partner_by_carrier

TDu opened the pull request:
The delivery slip can reference more than one sale order with different
customer references. If there is more than one of those references they
will be displayed in the lines.
But only one in the header!
So better not to display the customer reference in the header when
this is the case.

TDu committed:
[13.0][IMP] stock_picking_group_by_partner_by_carrier

The delivery slip can reference more than one sale order with different
customer references. If there is more than one of those references they
will be displayed in the lines.
But only one in the header, better not to display the customer reference
in the header with multiple sale order.
Also fix the customer reference to be specific to the delivery lines.

sebalix commented on stock_picking_group_by_partner_by_carrier/models/stock_picking.py:
```diff
@@ -139,6 +139,15 @@ def get_delivery_report_lines(self):
 
+    def get_customer_refs(self):
+        """Returns all unique sales order customer references."""
+        if self.state != "done":
+            move_lines = self.move_lines.filtered("product_uom_qty")
+        else:
+            move_lines = self.move_lines
```
Not sure to understand this trick?
There are moves without `product_uom_qty` filled? What is the link with the state of the picking?

simahawk commented:
@TDu can you squash?

TDu committed:
[13.0][IMP] stock_picking_group_by_partner_by_carrier

Remove the sentence about remaining items to be shipped.
As it is confusing with multiple sale order in a delivery

TDu commented:
I squashed the two first commits.
But was asked to add one more thing in this PR, that is the 2nd commit, now.

simahawk commented:
@TDu
```
2020-11-26 13:38:59,820 170 INFO openerp_test odoo.modules.loading: loading stock_picking_group_by_partner_by_carrier/report/report_delivery_slip.xml
2020-11-26 13:38:59,833 170 WARNING openerp_test odoo.addons.base.models.ir_ui_view: Error-prone use of @class in view report_delivery_document (): use the hasclass(*classes) function to filter elements by their classes
```

TDu commented:
@simahawk I fixed that warning with `hasclass`

simahawk commented:
/ocabot merge minor
